agent/engine/task_manager.go

```diff
@@ -53,6 +53,18 @@ const (
 	taskUnableToTransitionToStoppedReason = "TaskStateError: Agent could not progress task's state to stopped"
 )
 
+const (
+	// source for dockerContainerChange
+
+	// container transitions without any Docker actions, e.g. a `Created` container's desired status is set as `Stopped`
+	none ContainerChangeSource = iota
+	fromDockerEvent
+	fromInspect
+	fromDockerApi // from 'Docker start container' or 'Docker stop container'
```

Comment:
i think we inspect container after start container and stop container.. so i don't think you need a separate one for this?

Reply:
We do inspect for every start and stop calls, and also Docker events to get Metadata. Actually I'm handling `fromInspect` or `fromDockerAPI` the same way. But is it better to keep both so it's more clear what the source of the ContainerChange is?